doc/developers/develop.rst

```diff
@@ -723,6 +723,40 @@ method taking no input and returning a boolean. If this method exists,
 See :ref:`sphx_glr_auto_examples_developing_estimators_sklearn_is_fitted.py`
 for an example on how to use the API.
 
+Developer API for HTML representation
+=====================================
+
+.. warning::
+
+    The HTML representation API is experimental and the API is subject to change.
+
+Estimators inheriting from :class:`~sklearn.base.BaseEstimator` gain a functionality
+allowing to display an HTML representation of estimators in interactive programming
+environments such as Jupyter notebooks. For instance, we can display this HTML
+diagram::
+
+    from sklearn.base import BaseEstimator
+
+    BaseEstimator()
+
+The raw HTML representation can be obtained by invoking the function
+:func:`~sklearn.utils.estimator_html_repr` on an estimator instance. This representation
+is not intended to be tweak. The only supported use case is to properly redirect the
+user to the online documentation of the estimator when clicking on the "?" icon. While
+the defaults are suitable for scikit-learn, it is possible to customize the estimator
+such that it redirects to the intended webpage.
+
+It boils down to overriding the following attributes: (i) `_doc_link_module`, (ii)
```

Comment:
A suggestion for this paragraph, maybe combined with reducing the previous paragraph to just a sentence about the fact that `estimator_html_repr` exists (though I am not sure why we'd need to tell someone)

---

To customize the URL linking to an estimator's documentation override the `_doc_link_module` and `_doc_link_template` attributes. In addition you can provide a `_doc_link_url_param_generator` method.
Set `_doc_link_module` to the name of the (top level) module that contains your estimator. If the value
does not match the top level module name, the HTML representation will not contain a link to the
documentation. For scikit-learn estimators this is set to `"sklearn"`.

The `_doc_link_template` is used to construct the final URL. By default it can contain two variables:
`estimator_module` (the full name of the module containing the estimator) and `estimator_name` (the class name of the estimator). If you need more variables you should implement the `_doc_link_url_param_generator`
method which should return a dictionary of the variables and their values. This dictionary will be used
to render the `_doc_link_template`.

---

I changed the name of `_url_param_generator` to `_doc_link_url_param_generator`. It is a bit long now but it is consistent with prefixing the name with `_doc_link`. It isn't a great solution but I think it is better than having a free floating `_url_param_generator` which feels a bit too generic.